### PR TITLE
Added recursive SFTP copying

### PR DESCRIPTION
**Pull request by Caid11**
Added support for recursive copying to `copy_file`. Now, when a directory path is supplied to `local_file`, the method recursively calls itself on every file and directory within that directory, and attempts to copy them to the directory path supplied to `remote_file`.

### CODE REVIEW

File: pssh/ssh_client.py [hunk 1]
```diff
@@ -266,20 +266,27 @@ def copy_file(self, local_file, remote_file):
         :param remote_file: Remote filepath on remote host to copy file to
         :type remote_file: str
         """
-        sftp = self._make_sftp()
-        destination = remote_file.split(os.path.sep)
-        remote_file = os.path.sep.join(destination)
-        destination = destination[:-1]
-        for directory in destination:
+        if os.path.isfile(local_file):
```
Comment (pkittenis) [inline pssh/ssh_client.py]: What if `local_file` is a directory?

**Comment by pkittenis** [comment]
Thanks for the PR!

Am making changes to `copy_file` to fix #41 - will have to rebase on that once they're finished.

[This is the working branch](https://github.com/pkittenis/parallel-ssh/tree/mkdir_fix) for that WIP.

Couple comments in the meantime:

 * There should be a flag to turn recurse on/off - off by default (current behaviour)
 * Changes don't seem to handle `local_file` being a directory - see comment above.

If you want to rebase now, can assume that `mkdir` will recursively create sub-directories that don't exist - that will be covered as part of #41 .

**Comment by Caid11** [comment]
Thanks for the feedback! I'll make those changes and update the PR.

**Comment by Caid11** [comment]
`copy_dir` is now a separate function that calls copy file. Also, a new `recurse` flag has been added that prevents the function from moving through directories unless explicitly set to `True`.

File: pssh/ssh_client.py [hunk 2]
```diff
@@ -266,20 +283,21 @@ def copy_file(self, local_file, remote_file):
         :param remote_file: Remote filepath on remote host to copy file to
         :type remote_file: str
```
Comment (pkittenis) [inline pssh/ssh_client.py]: The new ``recurse`` keyword needs documenting here.

**Comment by Caid11** [comment]
I think that's a great way to do it. It makes it more clear what the `recurse` flag is doing for anyone reading the code, and makes the code a bit cleaner to look at.

I'll update the documentation and make that change. Thanks again for the feedback!

**Comment by pkittenis** [comment]
Awesome, thanks!

Have merged the `mkdir_fix` branch into master so should be able to rebase on master branch now (it's showing some conflicts here).

**Comment by Caid11** [comment]
Rebased `sftp_recursive` onto `parallel-ssh/master`. 

Sorry for the messy commit history; this is the first time  I've used git to do more than just push and pull.

**Comment by pkittenis** [comment]
Looks like a bad rebase + merge has created bad history - PR changes are showing changes from ``mkdir_fix`` branch and the PRs changes are missing.

Have re-created the PR at #45

**Pull request closed without merging**